src/MessageHandler.py

```diff
@@ -0,0 +1,112 @@
+import json
+import urllib2
+from src.MessageRouter import MessageRouter
+
+
+def new_message_handler(config):
+    def _new_command_handlers():
+        return {
+            'casts': _new_casts_handler(),
+            'streams': new_get_streams_response(fetch_url_contents),
+            'nyan': lambda message: ['~=[,,_,,]:3']
+        }
+
+    def _new_casts_handler():
+        return new_get_cats_response(
+            get_url_contents=fetch_url_contents,
+            blacklisted_youtubers=config['blacklisted_youtubers'] if 'blacklisted_youtubers' in config else [],
+            cast_count=int(config['cast_count']) if 'cast_count' in config else 5
+        )
+
+    message_router = MessageRouter(
+        command_handlers=_new_command_handlers(),
+        default_rate_limit_in_seconds=int(config['default_rate_limit']) if 'default_rate_limit' in config else 60
+    )
+
+    return message_router.handle_message
+
+
+def new_get_cats_response(get_url_contents, blacklisted_youtubers=[], cast_count=5):
+    def _get_videos_from_data(data):
+        return filter(_can_use_video, data['data']['items'])[:cast_count]
+
+    def _can_use_video(video):
+        return not _youtuber_is_blacklisted(video["uploader"])
+
+    def _youtuber_is_blacklisted(youtuber):
+        return youtuber in blacklisted_youtubers
+
+    def _get_cast_line(video):
+        date = video["uploaded"].split("T")[0]
+
+        like_count = video['likeCount'] if 'likeCount' in video else "0"
+
+        return "%s by %s - %s - %s (%s likes)" % \
+               (
+                   video['title'],
+                   video["uploader"],
+                   video['player']['default'].replace("&feature=youtube_gdata_player", ""),
+                   date,
+                   like_count
+               )
+
+    def get_cats_response(message):
```

Comment:
I think this needs to be more sophisticated, in that it will need to continually fetch videos if all the ones it gets are rejected by the blacklist. Sometimes the !casts output is full of spam, and with this it would only yield a single or no result.